internal/graph/check.go

```diff
@@ -1077,15 +1077,28 @@ func (c *LocalChecker) checkTTUSlowPath(ctx context.Context, req *ResolveCheckRe
 //
 // check(user, viewer, doc) will find the intersection of all group assigned to the doc's parent AND
 // all group where the user is a member of.
+
 func (c *LocalChecker) checkTTUFastPath(ctx context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset, iter *storage.ConditionsFilteredTupleKeyIterator) (*ResolveCheckResponse, error) {
 	ctx, span := tracer.Start(ctx, "checkTTUFastPath")
 	defer span.End()
 
+	typesys, ok := typesystem.TypesystemFromContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("typesystem missing in context")
+	}
+
+	terminalRelations := typesys.GetTerminalRelationsForTTUFastPath(
+		tuple.GetType(req.GetTupleKey().GetObject()), req.GetTupleKey().GetRelation(), tuple.GetType(req.GetTupleKey().GetUser()),
+	)
+	if len(terminalRelations) != 1 {
+		return nil, fmt.Errorf("expected exactly one terminal relation for fast path, received %d", len(terminalRelations))
+	}
```

Comment:
For now, direct and computed relations will only result in a single terminal relation. There should never be more than one terminal relation in the code's current state; this error helps catch this during testing. However, once support for the union and intersection rewrites is added, this will change.

Reply:
Similarly, what if we have something like 
```
type user1
type user2
type group1
  relations
    define member: [user1, user2]
type group2
type folder
  relations
    define owner: [group1, group2]
    define viewer: member from owner
```

Reply:
For the first scenario, we'll return `true` because both `group1#member` and `group2#member` usersets are fasth-path eligible. This is captured in the `multiple_ttu_references_to_multiple_types` test.

For the second, we'll return `true` because `group1#member` is fast-path eligible and `group2` lacks the `member` relation so we essentially skip it. This is captured in the `multiple_ttu_references_only_one_has_tupleset_relation` test.